framework/src/main/java/org/checkerframework/framework/source/SourceChecker.java

```diff
@@ -1449,6 +1458,12 @@ public boolean useUncheckedCodeDefault(String kindOfCode) {
         return false;
     }
 
+    /**
+     * Elements with a {@code @SuppressWarnings} that actually suppressed a warning for this
+     * checker.
+     */
+    public final Set<Element> elementsWithSuppressedWarnings = new HashSet<>();
```

Comment:
It seems very strange that some parts of this feature are implemented on the SourceChecker level and some on the BaseTypeChecker/BaseTypeVisitor level.
Wouldn't it be more consistent to pick the right level and implement the feature there?
You can have a SourceChecker subtype that doesn't use a BaseTypeVisitor subtype, for example.

Reply:
I wanted to implement it at the BaseTypeVisitor level, but I couldn't, because collecting elements with a `@SuppressWarnings` that actually suppressed a warning has to happen in the `Source Checker#shouldSuppressWarnings(javax.lang.model.element.Element, java.lang.String)`.  If the BaseTypeVisitor overrides that method and adds an element to `elementsWithSuppressedWarnings` if the super call returns true, then an element will be added for each recursive call to  `shouldSuppressWarnings`.   (Unless I've missed something here.)

I could implement at the Source Checker level, too, but the BaseType Checker would have to override that implementation to correctly implement it for compound checkers.

Reply:
Most warning management happens on the SourceChecker level, so that would seem like the right place to put all this logic.
Maybe you can add a method that can be overridden in BaseTypeChecker to handle compound checkers?